### PR TITLE
Mes 3665 pass debrief page extract components cat be

### PR DESCRIPTION
**Pull request by appleabs**
## Description
Components created for pass debrief page in preparation for category be.
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,22 +77,22 @@
...
     "cordova-plugin-autonomous-single-app-mode": "git+https://github.com/dvsa/cordova-plugin-autonomous-single-app-mode.git",
     "cordova-plugin-compat": "^1.2.0",
-    "cordova-plugin-device": "^2.0.2",
+    "cordova-plugin-device": "^2.0.3",
     "cordova-plugin-device-authentication": "git+https://github.com/dvsa/cordova-plugin-device-authentication.git",
     "cordova-plugin-google-analytics": "^1.8.6",
-    "cordova-plugin-inappbrowser": "^3.0.0",
-    "cordova-plugin-insomnia": "~4.3.0",
-    "cordova-plugin-ionic-keyboard": "^2.1.3",
+    "cordova-plugin-inappbrowser": "^3.1.0",
```
Comment (TomBillingtonUK) [inline package.json]: Can you please revert these changes and the changes to package-lock.json as it's changing the version number of the plugins we are using, which we shouldn't be doing

File: schema/schemaExaminerSchedule.json [hunk 2]
```diff
@@ -521,4 +521,4 @@
       "$ref": "#/definitions/deployments"
     }
   }
-}
+}
```
Comment (TomBillingtonUK) [inline schema/schemaExaminerSchedule.json]: Can you revert this change :)

File: src/modules/tests/test-data/test-data.cat-be.reducer.ts [hunk 3]
```diff
@@ -1,6 +1,5 @@
 import { TestData } from '@dvsa/mes-test-schema/categories/BE';
 import { createFeatureSelector, combineReducers, Action } from '@ngrx/store';
-
```
Comment (TomBillingtonUK) [inline src/modules/tests/test-data/test-data.cat-be.reducer.ts]: can you revert this change

File: src/pages/back-to-office/cat-b/back-to-office.cat-b.page.scss [hunk 4]
```diff
@@ -28,4 +28,4 @@ back-to-office-cat-b-page {
   }
 
 
-}
+}
```
Comment (TomBillingtonUK) [inline src/pages/back-to-office/cat-b/back-to-office.cat-b.page.scss]: can you revert this change

File: src/pages/debrief/cat-b/__tests__/debrief.cat-b.page.spec.ts [hunk 5]
```diff
@@ -1,7 +1,6 @@
 import { ComponentFixture, async, TestBed } from '@angular/core/testing';
 import { IonicModule, NavController, NavParams, Config, Platform } from 'ionic-angular';
 import { NavControllerMock, NavParamsMock, ConfigMock, PlatformMock } from 'ionic-mocks';
-
```
Comment (TomBillingtonUK) [inline src/pages/debrief/cat-b/__tests__/debrief.cat-b.page.spec.ts]: and this one :)

File: src/pages/pass-finalisation/cat-b/components/license-provided/__tests__/license-provided.spec.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    let component: LicenseProvidedComponent;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            declarations: [
+                LicenseProvidedComponent,
+            ],
+            imports: [
+                IonicModule,
+                TranslateModule,
```
Comment (TomBillingtonUK) [inline src/pages/pass-finalisation/cat-b/components/license-provided/__tests__/license-provided.spec.ts]: I don't think we need the Translate imports and providers as we aren't doing any translation on this field

File: src/pages/pass-finalisation/cat-b/components/license-provided/__tests__/license-provided.spec.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+    describe('Class', () => {
+        describe('ProvisionalLicenseReceived', () => {
+            it('should emit license received if license received is selected', () => {
+                spyOn(component.licenseReceived, 'emit');
+                component.provisionalLicenseReceived();
+                expect(component.licenseReceived.emit).toHaveBeenCalled();
+            });
+            
+            it('should emit license not received if license not received is selected', () => {
```
Comment (TomBillingtonUK) [inline src/pages/pass-finalisation/cat-b/components/license-provided/__tests__/license-provided.spec.ts:39]: this should be in a describe for provisionalLicenseNotReceived

File: src/pages/pass-finalisation/cat-b/components/pass-certificate-number/__tests__/pass-certificate.spec.ts [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+    let component: PassCertificateNumberComponent;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            declarations: [
+                PassCertificateNumberComponent,
+            ],
+            imports: [
+                IonicModule,
+                TranslateModule,
```
Comment (TomBillingtonUK) [inline src/pages/pass-finalisation/cat-b/components/pass-certificate-number/__tests__/pass-certificate.spec.ts]: Translate stuff should be able to go

File: src/pages/pass-finalisation/cat-b/components/transmission/__tests__/transmission.spec.ts [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    let component: TransmissionComponent;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            declarations: [
+                TransmissionComponent,
+            ],
+            imports: [
+                IonicModule,
+                TranslateModule,
```
Comment (TomBillingtonUK) [inline src/pages/pass-finalisation/cat-b/components/transmission/__tests__/transmission.spec.ts]: Translate stuff can go

File: src/pages/pass-finalisation/cat-b/components/transmission/__tests__/transmission.spec.ts [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+    describe('Class', () => {
+        describe('TransmissionChanged', () => {
+            it('should emit manual if manual transmission selected', () => {
+                spyOn(component.gearBoxCategoryChange, 'emit');
+                const transmission = 'Manual';
+                component.transmissionChanged(transmission);
+                expect(component.gearBoxCategoryChange.emit).toHaveBeenCalledWith(transmission);
+            });
+
+            it('should emit manual if manual transmission selected', () => {
```
Comment (TomBillingtonUK) [inline src/pages/pass-finalisation/cat-b/components/transmission/__tests__/transmission.spec.ts]: should be automatic not manual

File: src/pages/pass-finalisation/cat-b/_tests_/pass-finalisation.cat-b.page.spec.ts [hunk 11]
```diff
@@ -45,39 +68,77 @@ describe('PassFinalisationPage', () => {
...
-  describe('onSubmit', () => {
-    // Unit tests for the components TypeScript class
-    it('should dispatch the PersistTests action', () => {
-      const form = component.form;
-      form.get('provisionalLicenseProvidedCtrl').setValue(true);
-      form.get('passCertificateNumberCtrl').setValue('A123456*');
-      form.get('transmissionCtrl').setValue('Manual');
-      component.onSubmit();
-      expect(store$.dispatch).toHaveBeenCalledWith(new PersistTests());
+  describe('Class', () => {
```
Comment (TomBillingtonUK) [inline src/pages/pass-finalisation/cat-b/_tests_/pass-finalisation.cat-b.page.spec.ts:71]: missing tests for passCertificateNumberChanged()

File: src/pages/test-report/cat-b/__tests__/test-report.cat-b.page.spec.ts [hunk 12]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (TomBillingtonUK) [inline src/pages/test-report/cat-b/__tests__/test-report.cat-b.page.spec.ts]: can you revert the changes in this file

File: src/pages/waiting-room/cat-b/__tests__/waiting-room.cat-b.page.spec.ts [hunk 13]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room/cat-b/__tests__/waiting-room.cat-b.page.spec.ts]: can you revert the changes in this file

File: src/pages/waiting-room/cat-b/waiting-room.cat-b.page.scss [hunk 14]
```diff
@@ -18,4 +18,4 @@ waiting-room-cat-b-page {
       height: 18px;
     }
   }
-}
+}
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room/cat-b/waiting-room.cat-b.page.scss]: can you revert the changes in this file

File: src/pages/waiting-room/components/insurance-declaration/__tests__/insurance-declaration.spec.ts [hunk 15]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room/components/insurance-declaration/__tests__/insurance-declaration.spec.ts]: can you revert the changes in this file

**Review by TomBillingtonUK — APPROVED** [review]
Nice job, your nearly there. Just be careful on what you commit and push to avoid making uneeded changes to files.

**Review by MarkPatt89 — APPROVED** [review]
Looks good, just some conflicts to resolve.
